### ISSUE TITLE
Performance of PPG-based BP prediction

### ISSUE THREAD

**Issue by czy-orange** [Fabian-Sc85/non-invasive-bp-estimation-using-deep-learning#9]
Hi @Fabian-Sc85 , 
    Thanks for sharing the code. I ran the `ppg_training_mimic_iii.py` and drew the mae-epoch curves as fig1.png and fig2.png below, where the blue lines are for val set and orange lines for training set (detailed numbers are shown in the results.csv file).
    1. Are the observed results similar to yours? The mae for val set almost does not drop and is around 17/9 for SBP/DBP.
    2. Is it normal to see an exceptionally larger mae between epoch 30-35 and why?  
![fig1.png](https://user-images.githubusercontent.com/55686423/220834383-cd01e9c4-ae33-47e7-9555-8a773d37fee9.png)
![fig2.png](https://user-images.githubusercontent.com/55686423/220834391-f1585ceb-2791-45b9-8a7b-345b32971684.png)
[results.csv](https://github.com/Fabian-Sc85/non-invasive-bp-estimation-using-deep-learning/files/10810847/2023-21-02_resnet_my_pretraining_learningcurve.csv)

**Comment by Fabian-Sc85** [comment]
Hi @czy-orange,
yes, that is approximately the MAE values we have seen in our own work. You can compare your results to the results presented in section 4.1.3 of our Sensors paper (the link is in the Readme file). You can see that the network overfits almost immediately. This is because we use completely independent training, validation and test sets and don't allow mixing of subject-specific data across these sets.

Regarding the spike after epoch 30, I don't really know the reason for this. I have seen it before in my own training processes. It may be due to insufficient shuffling of the training samples between episodes because we have to read the training data from various tfrecord files.

**Comment by czy-orange** [comment]
@Fabian-Sc85 Thanks for your answer and explanation, which help me better understand the work and confirm the training process. Since the issue is solved, I'll close it.
